doc/man7/EVP_KDF-ARGON2.pod

```diff
@@ -0,0 +1,83 @@
+=pod
+
+=head1 NAME
+
+EVP_KDF-ARGON2 - The Argon2 EVP_KDF implementation
+
+=head1 DESCRIPTION
+
+Support for computing the B<argon2> password-based KDF through the B<EVP_KDF>
+API.
+
+The EVP_KDF-ARGON2 algorithm implements the Argon2 password-based key
+derivation function, as described in IETF Argon2 draft.  It is memory-hard in
+the sense that it deliberately requires a significant amount of RAM for efficient
+computation. The intention of this is to render brute forcing of passwords on
+systems that lack large amounts of main memory (such as GPUs or ASICs)
+computationally infeasible.
+
+=head2 Identity
+
+"ARGON2I", "ARGON2D", and "ARGON2ID" are the names for this implementation; it
+can be used with the EVP_KDF_fetch() function.
+
+=head2 Supported parameters
+
+The supported parameters are:
+
+=over 4
+
+=item "pass" (B<OSSL_KDF_PARAM_PASSWORD>) <octet string>
+
+=item "salt" (B<OSSL_KDF_PARAM_SALT>) <octet string>
+
+=item "secret" (B<OSSL_KDF_PARAM_SECRET>) <octet string>
+
+=item "iter" (B<OSSL_KDF_PARAM_ITER>) <unsigned integer>
+
+These parameters work as described in L<EVP_KDF(3)/PARAMETERS>.
+
+=item "t" (B<OSSL_KDF_PARAM_THREADS>) <unsigned integer>
+
+=item "s" (B<OSSL_KDF_PARAM_ARGON2_SZ>) <unsigned integer>
+
+=item "a" (B<OSSL_KDF_PARAM_ARGON2_AD>) <octet string>
+
+=item "l" (B<OSSL_KDF_PARAM_ARGON2_LANES>) <unsigned integer>
+
+=item "m" (B<OSSL_KDF_PARAM_ARGON2_MEMCOST>) <unsigned integer>
+
+=item "f" (B<OSSL_KDF_PARAM_ARGON2_FLAGS>) <unsigned integer>
+
+These parameters configure the Argon2.
+
+m is the memory size argument, l the number of lanes, a is the optional
+associated data, s is the digest length, t the number of threads.
+
+You may use f to control whether or not B<OPENSSL_cleanse> is called on
+password and/or secret during initial hash phase, by setting applying
+1 or 2 mask, respectively.
+
+t, s, l, m, and f are of type B<uint32_t>.
+
+=back
+
```

Comment:
Can you add sections for 
  NOTES
  EXAMPLES
  CONFORMING TO
  HISTORY

(Similiar to EVP_KDF_SS.pod)